spec/xattr_spec.cr

```diff
@@ -15,19 +15,30 @@ describe XAttr do
       File.delete(path)
     end
 
-    it "returns nil if xattr is not set" do
+    it "raises IO Error if xattr is not set" do
       path = "spec/test_get.txt"
       file = File.touch(path)
 
-      xattr = XAttr.new(path)
-      xattr["foo"].should eq nil
+      {% if flag?(:linux) %}
+        expect_raises(IO::Error, "Please check the target file: Operation not supported") do
+          xattr = XAttr.new(path)
+          xattr["foo"].should eq nil
```

Comment:
Wondering if this should just be `xattr["foo"]` with no expectations as it would raise an exception now.

Reply:
I suspect this is because xattr name not starts with `user.`.

`trusted.`, `system.` and other namespaces are reserved and need root access to set/get.

Reply:
Sorry. I misunderstood your comment. Fixed now

Reply:
There are a few differences between the multiple platforms. In Linux, if a xattr not exists then it returns `EPERM` if xattr is not having `user.` prefix else returns `ENODATA`. In darwin, if xattr not exists then it returns `ENOATTR`.